src/bindgen/ir/generic_path.rs

```diff
@@ -234,6 +255,48 @@ impl GenericArgument {
     }
 }
 
+/// Helper for erasing transparent types, which memoizes already-seen types to avoid repeated work.
+#[derive(Default)]
+pub struct TransparentTypeEraser {
+    // Remember paths we've already visited, so we don't repeat unnecessary work.
+    // TODO: how to handle recursive types such as `struct Foo { next: Box<Foo> }`?
+    known_types: HashMap<Type, Option<Type>>,
```

Comment:
Do we really expect extremely long chains of transparent typedefs? Memoizing this seems over-kill without performance numbers, IMHO...

Reply:
I don't know about long chains, but I do expect frequent use (typedefs exist to eliminate repetition in typing out complex type names). Is frequent use not sufficient reason to memoize?